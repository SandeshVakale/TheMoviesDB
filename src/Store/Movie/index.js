import {createAction, createReducer} from '@reduxjs/toolkit';

const initialState = {
  movie: {},
};

const setMovies = createAction('movies/setMovie');

export const movieModel = createReducer(initialState, builder => {
  builder.addCase(setMovies, (state, action) => {
    state.movie = action.payload;
  });
});
