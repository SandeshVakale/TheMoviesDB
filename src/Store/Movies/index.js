import {createAction, createReducer} from '@reduxjs/toolkit';

const initialState = {
  movies: [],
};

const setMovies = createAction('movies/setMovies');

export const moviesModel = createReducer(initialState, builder => {
  builder.addCase(setMovies, (state, action) => {
    state.movies = action.payload;
  });
});
