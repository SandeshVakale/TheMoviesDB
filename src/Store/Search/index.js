import {createAction, createReducer} from '@reduxjs/toolkit';

const initialState = {
  movies: [],
};

const setMovies = createAction('movies/setMovies');

export const searchModel = createReducer(initialState, builder => {
  builder.addCase(setMovies, (state, action) => {
    state.movies = action.payload;
  });
});
