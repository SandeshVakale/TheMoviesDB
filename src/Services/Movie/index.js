import {WAxios} from '../index';
import {store} from '../../Store';
export const getMovie = async movie_id => {
  let response;
  let errorRes;
  try {
    response = await WAxios.get(`movie/${movie_id}`);
    store.dispatch({
      type: 'movies/setMovie',
      payload: response.data,
    });
  } catch (error) {
    errorRes = error.response.data;
  }
  return response?.status === 200 ? true : errorRes;
};
