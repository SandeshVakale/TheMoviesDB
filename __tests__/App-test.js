import 'react-native';
import React from 'react';
import HomeScreen from '../src/Screens/Home';
import DetailScreen from '../src/Screens/Detail';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {movieModel} from '../src/Store/Movie';
jest.useFakeTimers();
jest.mock('react-native-elements', () => {
  return {
    SearchBar: () => {
      return <></>;
    },
    Text: () => {
      return <></>;
    },
    Card: () => {
      return <></>;
    },
    Divider: () => {
      return <></>;
    },
    Image: () => {
      return <></>;
    },
  };
});
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => jest.fn(),
}));
const mockStore = configureStore([]);

test('Home screen renders correctly', () => {
  const initialStore = {
    searchModel: {movies: []},
  };
  let store = mockStore(initialStore);
  const tree = renderer
    .create(
      <Provider store={store}>
        <HomeScreen />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Home screen renders correctly', () => {
  const initialStore = {
    searchModel: {
      movies: {
        total_results: 0,
        results: [
          {
            adult: false,
            backdrop_path: '/zJjI78peUFRIe2NPHthVhow2f60.jpg',
            genre_ids: [28, 18],
            id: 256040,
            original_language: 'te',
            original_title: 'బాహుబలి:ద బిగినింగ్',
            overview:
              'The young Shivudu is left as a foundling in a small village by his mother. By the time he’s grown up, it has become apparent that he possesses exceptional gifts. He meets the beautiful warrior princess Avanthika and learns that her queen has been held captive for the last 25 years. Shividu sets off to rescue her, discovering his own origins in the process.',
            popularity: 38.573,
            poster_path: '/9BAjt8nSSms62uOVYn1t3C3dVto.jpg',
            release_date: '2015-07-10',
            title: 'Bāhubali: The Beginning',
            video: false,
            vote_average: 7.5,
          },
        ],
      },
    },
  };
  let store = mockStore(initialStore);
  const tree = renderer
    .create(
      <Provider store={store}>
        <HomeScreen />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Detail screen renders correctly', () => {
  const initialStore = {
    movieModel: {movie: {}},
  };
  const route = {
    params: {
      id: 10,
      name: 'something',
    },
  };
  let store = mockStore(initialStore);
  const tree = renderer
    .create(
      <Provider store={store}>
        <DetailScreen route={route} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
