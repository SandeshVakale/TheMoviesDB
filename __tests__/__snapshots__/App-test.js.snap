// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail screen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</View>
`;

exports[`Home screen renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`Home screen renders correctly 2`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
